components/compiler-families/llvm-compilers/SPECS/llvm-compilers.spec

```diff
@@ -0,0 +1,191 @@
+Requires:      gnu-compilers%{PROJ_DELIM} = %{gnuver}
+%if 0%{?rhel_version} || 0%{?centos_version} || 0%{?rhel}
+BuildRequires: perl-Data-Dumper
+%endif
+
+%define install_path %{OHPC_COMPILERS}/llvm/%{version}
+%define gnu_path %{OHPC_COMPILERS}/gcc/%{gnuver}
+%if 0%{?sles_version} || 0%{?suse_version}
+%define compiler_path `%{OHPC_COMPILERS}/gcc/%{gnuver}/bin/g++ -v --version 2>&1|grep COMPILER_PATH|cut -f 2- -d '='`
+%define library_path `%{OHPC_COMPILERS}/gcc/%{gnuver}/bin/g++ -v --version 2>&1|grep LIBRARY_PATH|cut -f 2- -d '='`
+%endif
+
+%description
+
+The LLVM Compiler Infrastructure.
+
+%prep
+%setup -q -c -b1 -b2 -b3 -b4 -b5 -b6 -b7 -b8
+%patch1 -p0
+
+cd llvm-%{version}.src/tools
+ln -s ../../cfe-%{version}.src clang
+cd ../..
+
+cd llvm-%{version}.src/tools
+ln -s ../../lld-%{version}.src lld
+cd ../..
+
+cd llvm-%{version}.src/projects
+ln -s ../../compiler-rt-%{version}.src compiler-rt
+cd ../..
+
+cd llvm-%{version}.src/projects
+ln -s ../../libcxx-%{version}.src libcxx
+cd ../..
+
+cd llvm-%{version}.src/projects
+ln -s ../../libcxxabi-%{version}.src libcxxabi
+cd ../..
+
+cd llvm-%{version}.src/projects
+ln -s ../../libunwind-%{version}.src libunwind
+cd ../..
+
+cd llvm-%{version}.src/projects
+ln -s ../../openmp-%{version}.src openmp
+cd ../..
+
+ln -s llvm-%{version}.src llvm
+ln -s cmake-%{cmakever} cmake
+
+%build
+
+%install
+export OHPC_COMPILER_FAMILY=gnu
+. %{_sourcedir}/OHPC_setup_compiler
+
+mkdir build-cmake
+cd build-cmake
+../cmake/configure --prefix=%{_builddir}/cmake
+
+make %{?_smp_mflags} VERBOSE=1
+make install
+cd ..
+
+mkdir build-clang
+cd build-clang
+%{_builddir}/cmake/bin/cmake \
+--enable-optimise ../llvm \
+-DBUILD_SHARED_LIBS=True \
+-DCMAKE_BUILD_TYPE=Release \
+-DLIBOMP_FORTRAN_MODULES=False \
+-DLIBOMP_COPY_EXPORTS=False \
+-DLIBOMP_USE_HWLOC=False \
+-DLIBOMP_OMPT_SUPPORT=ON \
+-DLLVM_BINUTILS_INCDIR=/usr/include \
+-DLLVM_ENABLE_FFI=False \
+-DCMAKE_INSTALL_PREFIX=%{install_path}
+
+make %{?_smp_mflags} VERBOSE=1
+make DESTDIR=$RPM_BUILD_ROOT install
+cd ..
+
+# OpenHPC module file
+%{__mkdir_p} %{buildroot}/%{OHPC_MODULES}/llvm
+%{__cat} << EOF > %{buildroot}/%{OHPC_MODULES}/llvm/%{version}
+#%Module1.0#####################################################################
+
+proc ModulesHelp { } {
+
+puts stderr " "
+puts stderr "This module loads the LLVM compiler infrastructure"
+puts stderr " "
+
+puts stderr "\nVersion %{version}\n"
+
+}
+module-whatis "Name: LLVM Compiler Infrastructure"
+module-whatis "Version: %{version}"
+module-whatis "Category: compiler, runtime support"
+module-whatis "Description: LLVM Compiler Infrastructure"
+module-whatis "URL: http://www.llvm.org"
+
+set     version                            %{version}
+
+prepend-path    PATH                %{install_path}/bin
+prepend-path    MANPATH             %{install_path}/share/man
+prepend-path    INCLUDE             %{install_path}/include
+prepend-path    LD_LIBRARY_PATH     %{gnu_path}/lib64:%{install_path}/lib
```

Comment:
Most of the LLVM and Clang is writen in C++, if we build it with non-distro compiler, it needs to have this path to libs provided.